src/FastRoute.php

```diff
@@ -252,6 +260,83 @@ public function generate(string $name, array $parameters = []): string
         ));
     }
 
+    /**
+     * Generates absolute URL from named route and parameters
+     *
+     * @param string $name name of the route
+     * @param array $parameters parameter-value set
+     * @param string|null $scheme host scheme
+     * @param string|null $host host for manual setup
+     * @return string URL generated
+     * @throws RouteNotFoundException in case there is no route with the name specified
+     */
+    public function generateAbsolute(string $name, array $parameters = [], string $scheme = null, string $host = null): string
```

Comment:
I suggest adding the schema and host inside the parameters

Reply:
The last two parameters are optional, so in most cases we use syntax like
```php
generateAbsolute('route-name', ['id' => 42]);
```

Reply:
I'd not do that. There could be parameters named `host` and `scheme`.

Reply:
makes sense